pandas/core/generic.py

```diff
@@ -3791,7 +3796,10 @@ def values(self):
         int32. By numpy.find_common_type convention, mixing int64 and uint64
         will result in a flot64 dtype.
         """
-        return self.as_matrix()
```

Comment:
just return ``.values``

Reply:
This *is* the def of `.values`

Reply:
oh, then we should push this entirely down to the BlockManager, so pass the axis as well. The BM can then do the transpose, we don't like to do things in user code like this.

Reply:
We should actually deprecate ``.values`` as well, in favor of ``.to_numpy()`` which is the future spelling anyhow.

Reply:
This is already almost entirely pushed down, since it is calling directly BlockManager.as_matrix (but OK, the transpose could be done inside `BlockManager.as_matrix`). The `consolidate_inplace` needs to happens in NDFrame I think.

Deprecating `.values` is a no-go IMO, but let's deprecate that in another issue if you want

Reply:
> oh, then we should push this entirely down to the BlockManager, so pass the axis as well. The BM can then do the transpose, we don't like to do things in user code like this.

I'm not too familiar with the BlockManager, except callin it with ``._data``. How do I pass the axis to it, and how do I transpose in a BlockManager? E.g. it doesn't have a ``.transpose`` method.

Reply:
look at ``_take`` in ``pandas/core/generic.py``, you just pass the ``get_block_manager_axis`` to it. 

then add ``axis=`` to ``def as_matrix`` in ``pandas/core/internals`` and do the transpose there. 

The reason for pushing this to internals is to avoid all of the wrapping layers (e.g. frame/series, etc.) to know about internal details like this.

Reply:
``as_matrix`` doesn't have a axis parameter. Do you mean adding a new axis parameter there and transposing inside, if ``axis=0`` (where I assume from my limited knowledge that ``BlockManager.axes[0]`` is always the same as ``dataFrame.axes[1]``, correct?)

Reply:
Hi @jreback, I've tried quite a bit today and I can't push this down while getting the tests to pass.

Could you look at my latest commit (not passing ATM) and give some advice?